### PR TITLE
feature/creating contact list screen

### PR DESCRIPTION
**Pull request by dtngoc99**
## Change Description 👋

Creating contact list screen with some actions such as:

- Show contact list
- Change customer status and supplier status
- Delete contact
- Search contact
- Filter contact by status

## Demo 📺

https://loom.com/share/74dc8ab43982436dbcbe3ba9d36ef559

### CODE REVIEW

**Comment by dtngoc99** [comment]
Update demo:

https://loom.com/share/639b3e7c67f64f508b00a34682a97094

**Comment by luantc13** [comment]
:stuck_out_tongue_winking_eye:
image

File: src/contacts/components/Contacts.vue [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+  methods: {
+    confirmChangeStatus(contact, contactTypeChange) {
+      const actionTypeName =
+        contactTypeChange === CUSTOMER_TYPE
+          ? contact.customerStatus === ENABLE_STATUS
+            ? "disable customer status"
+            : "enable customer status"
+          : contact.supplierStatus === ENABLE_STATUS
+          ? "disable supplier status"
+          : "enable supplier status";
```
Comment (luantc13) [inline src/contacts/components/Contacts.vue:211]: Pls, don't use this magic anymore, `if else` is the best choice here.

File: src/contacts/store/index.js [hunk 2]
```diff
@@ -0,0 +1,114 @@
+import {
+  getContacts,
+  updateContactStatus,
+  deleteContact,
+} from "@/http/contacts.js";
+import Vue from "vue";
+import NProgress from "nprogress";
+const namespaced = true;
+const state = {
```
Comment (luantc13) [inline src/contacts/store/index.js:9]: Empty lines to break the code to separate blocks are easier to read in the future.
An example, we can break this file into 6 blocks: `import`, `state`, `mutations`, `actions`, `getters`, and `export`.
Each block is separated by an empty line.

File: src/contacts/store/index.js [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+    state.pagination = {
+      ...state.pagination,
+      ...pagination,
+    };
+  },
+  SET_CONTACT_STATUS(state, { contactId, contactStatus, contactTypeChange }) {
+    const index = state.contacts.findIndex((c) => c.id === contactId);
+    if (contactTypeChange === 0)
+      state.contacts[index].customerStatus = contactStatus;
+    else state.contacts[index].supplierStatus = contactStatus;
```
Comment (luantc13) [inline src/contacts/store/index.js:31]: We should have `{}` for all `if else` statements, it's a javascript convention.
```suggestion
    if (contactTypeChange === 0) {
      state.contacts[index].customerStatus = contactStatus;
    } else {
      state.contacts[index].supplierStatus = contactStatus;
    }
```

File: src/http/contacts.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+      value: contactStatus,
+      path: `${contactTypePath}`,
+      op: "replace",
+    },
+  ]);
+};
+const deleteContact = ({ companyId, contactId }) => {
+  return httpClient.delete(`${COMPANY}/${companyId}${CONTACTS}/${contactId}`);
+};
+export { getContacts, updateContactStatus ,deleteContact};
```
Comment (luantc13) [inline src/http/contacts.js:22]: Not a good formating.

File: src/http/endpoints.js [hunk 5]
```diff
@@ -12,3 +12,4 @@ export const COMPANY_ENDPOINT = "/companies";
 export const PRODUCTS = "/products";
 export const IMAGE = "/image";
 export const DELETE_COMPANY_ENDPOINT = "/companies";
+export const CONTACTS='/contacts'
```
Comment (luantc13) [inline src/http/endpoints.js:15]: All files should have an empty line at the end.

File: src/plugins/nprogress.js [hunk 6]
```diff
@@ -0,0 +1,3 @@
+import NProgress from "nprogress";
+import "nprogress/nprogress.css";
+NProgress.configure({showSpinner:false})
```
Comment (luantc13) [inline src/plugins/nprogress.js:3]: All files should have an empty line at the end.

File: src/products/store/index.js [hunk 7]
```diff
@@ -36,6 +36,11 @@ const mutations = {
   UPDATE_PRODUCT_IMAGE(state, data) {
     state.product.imageName = data.imageName;
   },
+  CLEAR_ALL_STATES(){
+    state.product=undefined;
+    state.products=[];
+    state.totalPages=0;
```
Comment (luantc13) [inline src/products/store/index.js:42]: Not a good formating.

File: src/router/index.js [hunk 8]
```diff
@@ -221,8 +224,10 @@ router.beforeEach((routeTo, routeFrom, next) => {
   ) {
     next({ path: "/auth/signin" });
   } else {
+    store.dispatch("contacts/clearAllStates");
+    store.dispatch("employees/clearAllStates");
+    store.dispatch("products/clearAllStates");
```
Comment (luantc13) [inline src/router/index.js:229]: I don't think that is a good idea when clearing all states `router.beforeEach`.
Just clear the necessary fields is too enough.

File: src/store/index.js [hunk 9]
```diff
@@ -25,5 +26,6 @@ export default new Vuex.Store({
     forgotPassword,
     resetPassword,
     products,
+    contacts
   },
 });
```
Comment (luantc13) [inline src/store/index.js:31]: All files should have an empty line at the end.

**Review by luantc13 — CHANGES_REQUESTED** [review]
Great work!
Having some comments, happy to merge when they are resolved.

File: src/contacts/store/index.js (hunk 2 above)
Comment (dtngoc99) [reply src/contacts/store/index.js:9]: I will

**Review by luantc13 — APPROVED** [review]
Great work!

**Review by truonghungit — APPROVED** [review]

**Pull request merged**